src/context.rs

```diff
@@ -352,6 +353,18 @@ impl Context {
     pub fn tests(&self) -> &HashMap<CtxKey, FunctionDec> {
         &self.tests
     }
+
+    pub fn eval(&mut self, input: &String) -> Result<Option<ObjectInstance>, Error> {
+        parser::parse(self, input)?;
+
+        self.emit_errors();
+        self.clear_errors();
+
+        let res = self.execute()?;
```

Comment:
Sadly this is gonna fail typechecking as the type context is recreated at each call to `self.execute()`

Reply:
This as been fixed by #436